Change return type of signMessage

cemozerr opened the pull request.

cemozerr commented on packages/wallets/alpha/src/adapter.ts:
```diff
@@ -218,14 +218,14 @@ export class AlphaWalletAdapter extends BaseMessageSignerWalletAdapter {
     }
 
-    async signMessage(message: Uint8Array): Promise<Uint8Array> {
+    async signMessage(message: Uint8Array): Promise<{signature: Uint8Array, signedMessage: Uint8Array}> {
```
Please[ read the discussion here.](https://github.com/anza-xyz/wallet-adapter/pull/1094#issuecomment-2937571940)

cemozerr replied:
Happy to change it if I digested the discussion wrong, but that is my understanding.

jordaaash replied:
This is correct. `signedMessage` is the message that we're asserting got signed, which may be different than the message provided. Of course, since we're wrapping wallets that may do whatever before signing, we're just guessing in our adapters until they update their implementations. But this gets us where we want with Ledger and with the wallets that already support the Wallet Standard, since they already use this interface.

jordaaash replied:
@beeman FWIW the intent here is to make the API of this [the same as the Wallet Standard](https://github.com/anza-xyz/wallet-standard/blob/4f6248e90581f0032553eb4dbdd8bdee65342f8e/packages/core/features/src/signMessage.ts#L36-L54).

@cemozerr I wonder if, to that effect, we can just return the `SolanaSignMessageOutput` type?

jordaaash replied:
(The `signatureType` right now does nothing, but the intent was to allow things like passkeys to sign the message with secp256r1. I suspect we'll see this get used by Squads now that the secp256r1 precompile is live.)

cemozerr commented:
@jordaaash Please check [the latest commit](https://github.com/anza-xyz/wallet-adapter/pull/1106/commits/23a450ee0685a8b6fa440e59ca9554b603b54044) - I believe I've addressed all your comments.

ianleck commented:
@jordaaash Hi just to check in - will we be supporting signedMessage being returned in the wallet adapter, since this is required to support ledger's OCMS? I recently merged in a PR on ledger that fixes it - now we just need wallet adapter support for dapps to be able to fully utilise ocms on ledgers
